### DISCUSSION TITLE
UI design

### DISCUSSION THREAD

**Discussion by Nexucis** [perses/perses#185] [Design Discussions]
Hello,

I started a draft for the design of the Perses' UI. For the moment I only did the Dashboard view

## Single Dashboard view

![image](https://user-images.githubusercontent.com/4548045/143236503-1b0c4c05-2817-4067-a988-c8a4ecc86f34.png)

## Single Dashboard view with scrolling

![image](https://user-images.githubusercontent.com/4548045/143236562-4cb596c8-7b75-454c-b29a-5350d9cdce9c.png)

Let me know what you think about it !

Note: I used [pencil](https://github.com/evolus/pencil) to do it. I can shared the folder if someone want to edit it

**Comment by roidelapluie** [comment]
Thanks!

I don't believe in variables on the side because they can be quite long (typical usage: FQDN). If you only have one or two variables, you would lose a lot of space.

**Reply by Nexucis** [reply]
it's true indeed.
In the other side having it on a side bar will avoid you to go to the top each time you want to change the variable's value.
Probably the left bar can be scrollable to handle the case when you have many different variables.

Do you think with a scrolling bar it would do it @roidelapluie ?

**Reply by roidelapluie** [reply]
No, I would rather have the variables bar stick to the top of the screen if we need to.

**Reply by AntoineThebaud** [reply]
Agree, + imho it's more intuitive in the (common) case where variables are influencing each other in order.

**Reply by Nexucis** [reply]
And like this ? : 

![image](https://user-images.githubusercontent.com/4548045/143281395-63a456b6-6a5d-4766-a111-aba924a1d18d.png)

![image](https://user-images.githubusercontent.com/4548045/143281474-3c80f182-0b09-4b15-8e01-53fc6db3c9a8.png)

The idea is that the top bar that contains the logo is not fixed, but the place where the variables are displayed is still present even when you are scrolling.
Of course you will have a button to hide this top bar in case you would like more places to display the different graphs.

**Reply by jhardy** [reply]
I'm curious if we have some realistic examples of long strings that would be displayed. I am starting to put together a few additional ideas for variable layout and any realistic examples of long strings would be helpful. For example, it would be great to have a few realistic `FQDN` that we wold need to accommodate.

**Reply by Nexucis** [reply]
The example that came to my mind is the list of hostname of the machine. They can be pretty long:
![image](https://user-images.githubusercontent.com/4548045/156589793-c7994a78-a286-482a-ad3b-3d6935e57dd7.png)

Or you could have the path of the internal organisation. That helps to know who is the owner of what for example

**Reply by Nexucis** [reply]
Like PPT-TEC-OOS-OSV-OYA-JHD-LSN

**Comment by Nexucis** [comment]
Adding more design a bit focused around the navigation.
Note: it won't represent the final rendering of the UI. They are just schemas here that I made to share ideas.

### BreadCrum for sub-folder
![image](https://user-images.githubusercontent.com/4548045/144621965-977d6b89-fa86-4369-9e4b-264e113cfdef.png)

### BreadCrum for Dashboards
![image](https://user-images.githubusercontent.com/4548045/144622069-262c1293-b15b-4b6f-921d-ee67c9d51c73.png)

### Projects/Dashboard List view
Nothing too fancies here. Just a way to display all projects available, and then when you expend a line, you should see the folder tree with the dashboards inside.


![image](https://user-images.githubusercontent.com/4548045/144622155-a976c16c-3f72-41be-aa36-3ef25d99318c.png)

### Single Project view

![image](https://user-images.githubusercontent.com/4548045/144622275-95546d2f-88f3-4786-bc3e-b109f6d5035f.png)

**Reply by celian-garcia** [reply]
My 2 cent here :)
- Why not integrating the project in the breadcrumb, as the UI lacks a way to know in which project we are (apart from the url of course)
- What would you put in the expand of projects list ? why not just a simple list
- I would expect the project view to be more complex, with yes the folder architecture, but also on top, the project name, (breadcrumb would be enough?), who created the project, some tags, a description of the project...

**Reply by celian-garcia** [reply]
Thanks for these view btw, it helps me in the direction I would go for the project CRUD, even if everything is quite open I guess ^^. Let's discuss on Monday!

**Reply by Nexucis** [reply]
actually the design moved to another discussion: https://github.com/perses/perses/discussions/792. You should take a look at this.
